Explorer/Assets/Scripts/Global/Dynamic/RealmController.cs

```diff
@@ -86,9 +90,44 @@ public async UniTask SetRealmAsync(URLDomain realm, CancellationToken ct)
             if (!ComplimentWithStaticPointers(world, realmEntity) && !realmComp.ScenesAreFixed)
                 ComplimentWithVolatilePointers(world, realmEntity);
 
-            teleportController.OnRealmLoaded(realmData.ScenesAreFixed ? retrieveSceneFromFixedRealm : retrieveSceneFromVolatileWorld);
+            IRetrieveScene sceneProviderStrategy = realmData.ScenesAreFixed ? retrieveSceneFromFixedRealm : retrieveSceneFromVolatileWorld;
+            sceneProviderStrategy.World = globalWorld.EcsWorld;
+            teleportController.SceneProviderStrategy = sceneProviderStrategy;
 
-            // Hide loading screen
+            var sceneLoadReport = new AsyncLoadProcessReport(new UniTaskCompletionSource(), new AsyncReactiveProperty<float>(0));
+
+            if (loadReport != null)
+                PropagateSceneLoadReportToRealmLoadReportAsync(loadReport.ProgressCounter.Value, loadReport, sceneLoadReport, ct).Forget();
```

Comment:
You can merge it via `UniTask.WhenAll` with `await teleportController.TeleportToSceneSpawnPointAsync(playerStartPosition, sceneLoadReport, ct);`